packages/flutter/test/material/popup_menu_test.dart

```diff
@@ -851,6 +851,139 @@ void main() {
     expect(tester.getTopLeft(find.text('Item 0')).dx, 72);
     expect(tester.getTopLeft(find.text('Item 1')).dx, 72);
   });
+
+  testWidgets('PopupMenuButton takes only child or only icon', (WidgetTester tester) async {
+    // [PopupMenuButton] should assert that either child == null, icon == null,
+    // or that both are null because only one of the two can be displayed/used.
+    expect(PopupMenuButton<int>(
+      itemBuilder: (BuildContext context) => <PopupMenuItem<int>>[],
+      child: Container(),
+      icon: const Icon(Icons.error),
+    ), throwsA(const TypeMatcher<AssertionError>()));
```

Comment:
@shihaohong This does not work. Is it even possible to test assertions?

Reply:
https://cirrus-ci.com/task/6552971109728256?command=test#L2956

Reply:
Try something like https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/chip_test.dart#L1815

Reply:
Right, thanks.